src/imagecapturecore.cs

```diff
@@ -0,0 +1,900 @@
+		[Export ("deviceBrowser:didAddDevice:moreComing:")]
+		void DidAddDevice (ICDeviceBrowser browser, ICDevice device, bool moreComing);
+
+		[Abstract]
+		[Export ("deviceBrowser:didRemoveDevice:moreGoing:")]
+		void DidRemoveDevice (ICDeviceBrowser browser, ICDevice device, bool moreGoing);
+
+		[Export ("deviceBrowser:deviceDidChangeName:")]
+		void DeviceDidChangeName (ICDeviceBrowser browser, ICDevice device);
+
+		[Deprecated (PlatformName.MacOSX, 10, 13)]
+		[Export ("deviceBrowser:deviceDidChangeSharingState:")]
+		void DeviceDidChangeSharingState (ICDeviceBrowser browser, ICDevice device);
+
+		[Export ("deviceBrowser:requestsSelectDevice:")]
+		void RequestsSelectDevice (ICDeviceBrowser browser, ICDevice device);
+
+		[Export ("deviceBrowserDidEnumerateLocalDevices:")]
+		void DidEnumerateLocalDevices (ICDeviceBrowser browser);
+	}
+
+	[BaseType (typeof(NSObject))]
+	interface ICDeviceBrowser {
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		IICDeviceBrowserDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Assign)]
+		NSObject WeakDelegate { get; set; }
+
+		[Export ("browsing")]
+		bool Browsing { [Bind ("isBrowsing")] get; }
+
+		[Export ("browsedDeviceTypeMask", ArgumentSemantic.Assign)]
+		ICBrowsedDeviceType BrowsedDeviceTypeMask { get; set; }
+
+		[NullAllowed, Export ("devices")]
+		ICDevice[] Devices { get; }
+
+		[NullAllowed, Export ("preferredDevice")]
+		ICDevice PreferredDevice { get; }
+
+		[Export ("start")]
+		void Start ();
+
+		[Export ("stop")]
+		void Stop ();
+	}
+
+	[BaseType (typeof(NSObject))]
+	interface ICCameraItem {
+
+		[Export ("device")]
+		ICCameraDevice Device { get; }
+
+		[Export ("parentFolder")]
+		ICCameraFolder ParentFolder { get; }
+
+		[Export ("name")]
+		string Name { get; }
+
+		[Export ("UTI")]
+		string Uti { get; }
+
+		[NullAllowed, Export ("fileSystemPath")]
+		string FileSystemPath { get; }
+
+		[Export ("locked")]
+		bool Locked { [Bind ("isLocked")] get; }
+
+		[Export ("raw")]
+		bool Raw { [Bind ("isRaw")] get; }
+
+		[Export ("inTemporaryStore")]
+		bool InTemporaryStore { [Bind ("isInTemporaryStore")] get; }
+
+		[NullAllowed, Export ("creationDate")]
+		NSDate CreationDate { get; }
+
+		[NullAllowed, Export ("modificationDate")]
+		NSDate ModificationDate { get; }
+
+		[NullAllowed, Export ("thumbnailIfAvailable")]
+		CGImage ThumbnailIfAvailable { get; }
+
+		[NullAllowed, Export ("largeThumbnailIfAvailable")]
+		CGImage LargeThumbnailIfAvailable { get; }
+
+		[NullAllowed, Export ("metadataIfAvailable")]
+		NSDictionary<NSString, NSObject> MetadataIfAvailable { get; }
+
+		[NullAllowed, Export ("userData")]
+		NSMutableDictionary UserData { get; }
+
+		[Export ("ptpObjectHandle")]
+		uint PtpObjectHandle { get; }
+
+		[Export ("addedAfterContentCatalogCompleted")]
+		bool AddedAfterContentCatalogCompleted { [Bind ("wasAddedAfterContentCatalogCompleted")] get; }
+	}
+
+	[BaseType (typeof(ICCameraItem))]
+	interface ICCameraFolder {
```

Comment:
double check if this is user-creatable (for .ctor)
maybe it's `ICCameraItem` that is abstract ?